modules/core/04-channel/types/upgrade_test.go

```diff
@@ -94,30 +94,22 @@ func (suite *TypesTestSuite) TestUpgradeErrorIsOf() {
 		{
 			msg: "not equal to nil error",
 			malleate: func() {
-				upgradeError = types.UpgradeError{}
+				upgradeError = &channeltypes.UpgradeError{}
 			},
 			expPass: false,
 		},
 		{
 			msg: "wrapped upgrade error",
 			malleate: func() {
 				wrappedErr := errorsmod.Wrap(upgradeError, "wrapped upgrade error")
-				upgradeError = types.NewUpgradeError(1, wrappedErr)
-			},
-			expPass: true,
-		},
-		{
-			msg: "nil underlying error and target",
-			malleate: func() {
-				upgradeError = types.UpgradeError{}
-				intputErr = types.UpgradeError{}
```

Comment:
this test case was removed, because we don't actually need to compare a nil error to see if it's a nil error.

The `errors.Is` check in stdlib does not actually return true if two nil errors are both nil but are of different implementations. E.g. the following test function does not pass

```golang
func TestFoo(t *testing.T) {
   var err1 *channeltypes.UpgradeError = nil
   var err2 error = nil
   require.Nil(t, err1)
   require.Nil(t, err2)
   res := errorsmod.IsOf(err1, err2)
   require.True(t, res)
}
```